### PR TITLE
Import work load data

### PR DESCRIPTION
**Pull request by jdtemple**
- adds functions for several dimension and fact table csv imports
- adds DbUp project for easier database management in a development environment

The DbUp project includes the ability to supply a "rebuild" command line argument that will wipe your db and build you a new one from scratch. This is a carryover from modifications I have made to our in-house .NET Core project template.

I think this may include some earlier API work from @Rhipeus since the starting branch was a WIP progress branch of his.

### CODE REVIEW

File: source/WesternStatesWater.WaDE.DbUp/Scripts/002 Table Types.sql [hunk 1]
```diff
@@ -0,0 +1,226 @@
+/****** Object:  UserDefinedTableType [Core].[AggregatedAmountTableType]    Script Date: 5/2/2019 11:13:30 AM ******/
```
Comment (Rhipeus) [inline source/WesternStatesWater.WaDE.DbUp/Scripts/002 Table Types.sql]: Let's put these files into a `Scripts/001/` folder.

We have found in Westlake that structure allows us more flexibility.  Then we go to a new folder for each release.

File: source/WesternStatesWater.WaDE.DbUp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        }
+
+        static void Main(string[] args)
+        {
+            var connectionString = Configuration["WadeDatabase"];
+
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new InvalidOperationException("Connection string environment variable missing.");
+            }
```
Comment (Rhipeus) [inline source/WesternStatesWater.WaDE.DbUp/Program.cs]: For this, I have used a bit of code that adds a bit more flexibility/safety.  A connection string can be used as an argument and we confirm rebuild (unless we are non-interactive or a force flag is used).  This tends to save our bacon from time to time.  This *isn't* exactly what you want but it should be close.

``` C#
[STAThread]
        static void Main(string[] args)
        {
            InitializeAzureKeyVaultProvider();
            (var connectionString, var rebuild, var force) = ParseParameters(args);

            EnsureDatabase.For.SqlDatabase(connectionString);

            if (rebuild)
            {
                RebuildDatabase(connectionString, force);
            }

            UpdateDb(connectionString);

            Console.WriteLine("Success!");
        }

        private static (string ConnectionString, bool Rebuild, bool Force) ParseParameters(string[] args)
        {
            var connectionString = ConfigurationManager.ConnectionStrings["ApplicationServices"].ConnectionString;
            var rebuild = false;
            var force = false;
            if (args != null && args.Length > 0)
            {
                rebuild = args[0] == "rebuild";
                if (args.Length > 1)
                {
                    connectionString = args[1];
                    if (args.Length > 2)
                    {
                        force = args[2] == "force";
                    }
                }
            }
            return (connectionString, rebuild, force);
        }

        private static void RebuildDatabase(string connectionString, bool force)
        {
            bool doCleanup = true;
            if (!force && Environment.UserInteractive)
            {
                // ask user if they really want this
                Console.WriteLine("###User Interactive Mode Question###");
                Console.WriteLine("Do you want to rebuild the database? yes/no");
                var yesNo = Console.ReadLine();
                if (!string.IsNullOrWhiteSpace(yesNo) && yesNo.ToLower() == "yes")
                    doCleanup = true;
                else
                    doCleanup = false;

                Console.WriteLine($"doCleanup: {doCleanup}");
            }

            if (doCleanup)
                CleanupTables(connectionString);
        }
```

File: source/WesternStatesWater.WaDE.DbUp/Scripts/002 Table Types.sql (hunk 1 above)
Comment (jdtemple) [reply source/WesternStatesWater.WaDE.DbUp/Scripts/002 Table Types.sql]: I totally forgot the folder!

Comment (jdtemple) [reply source/WesternStatesWater.WaDE.DbUp/Scripts/002 Table Types.sql]: This is done.

File: source/WesternStatesWater.WaDE.DbUp/Program.cs (hunk 2 above)
Comment (jdtemple) [reply source/WesternStatesWater.WaDE.DbUp/Program.cs]: This is done.

**Pull request merged**